### ISSUE TITLE
Ubuntu 19.10 installation issues

### ISSUE THREAD

**Issue by anlag** [ja2-stracciatella/ja2-stracciatella#886]
I stumbled on this project today and decided to give it a go. Armed with a freshly bought discounted copy of JA2 from GOG.com, I attempted to install it on my Ubuntu 19.10 desktop. I had a few issues along the way which I was able to get around, but wanted to note them down here. I realize strictly speaking I should open a separate issue for each one, but I'm a bit reluctant to spam the issue tracker in the first instance when it may just have been me missing something.

Nevertheless, here are the issues I encountered and how I resolved them.

## libpng12 issue
After installing the latest stable version (0.16.1) and trying to execute `ja2-launcher`, I got a library error:

```
ja2-launcher: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
```
Searching issues I found this to be [fixed](https://github.com/ja2-stracciatella/ja2-stracciatella/issues/747) in nightlies, so I installed version 0.16.x-20191012 (subsequent deb builds are 0 bytes) which resolved this problem.

## libfltk_images issue
Having got past the libpng issue, I ran into another missing library.
```
ja2-launcher: error while loading shared libraries: libfltk_images.so.1.3: cannot open shared object file: No such file or directory
```
Unlike the first issue, I was unable to find this referenced in the issue tracker. It was easy enough to resolve however, by simply installing `libfltk-images1.3` from the distribution repositories (`sudo apt install libfltk-images1.3` for completeness.) With this done, I was able to run the launcher.

## Data dir not found
With the launcher running, I needed to specify the game directory, which gave me some pause since all I had was a Windows executable for the installer. The only way I could think of proceeding was to install Wine (`sudo apt install wine`) and then install it through there (`wine setup_jagged_alliance_2_1.12_\(17794\).exe`). The installation crashed with some errors but only after the game directory had been created so I didn't try to resolve them.

Upon specifying the (default) game directory (`~/.wine/drive_c/GOG Games/Jagged Alliance 2`) in the launcher, I received a "Data dir not found" error. I didn't find this specifically addressed in the issue tracker but another issue made me wonder if the spaces in the path might be the cause, so I tried copying the directory to `~/Apps/JA2`. When entering that into the launcher, the game started up.

## Resolution
I can change the resolution in the launcher and the game window does change accordingly, however the actual game content remains at the default 640x480 size, meaning in a larger window the game is surrounded by a lot of black.

I'm able to get the actual game to scale up in size by opening it in a 640x480 window and then resizing the window, which causes the game to scale up in size too. Nevertheless I don't think this is the intended behaviour? I would expect if entering for example 1024x768 that the game fills a window of that size upon launch.

## Linux instructions
It might be helpful to include in the "How to run" page a note that Linux users with the Windows installer (I suspect more people have this than the old Linux port) may wish to conduct a Wine install to obtain the game directory. If any of the other issues above are known obstacles, it might be helpful to point them out too.

---

Apologies for the verbosity. Should I file separate issues for the above stumbling blocks (except libpng which is already there, albeit not in the stable version) or is this already known and on the cards despite my not finding them on the site?

Lastly many thanks for this project, I look forward to digging into the game. Like many others I played the original (JA1) in the 90s and loved it, hoping the sequel holds up as well as most indicate despite the fact it's already two decades old.

**Comment by lynxlynxlynx** [comment]
Hey, the builds are not static, so they will always fail if the underlying system diverges too much. That's why distribution packages are preferred and it's another good reason to have releases more often.

Higher resolution is only visible for the tactical view, since there are no alternative graphics for the bigger sizes. JA2S needs the original game data and we don't care how you get it (there're several options). It was always deemed a simple exercise left to the reader.

The concise list of problems I see:
- the list of dependencies in COMPILATION.md is not complete
- (launcher) mishandles paths with spaces?
- docs updates for new stuff in the next release as noted in #654

**Comment by flaviojs** [comment]
The paths with spaces problem should be fixed with #888.
